src/chain_spec.rs

```diff
@@ -123,6 +124,7 @@ impl LightSyncState {
 #[derive(Clone)]
 pub struct ChainSpec {
     client_spec: structs::ClientSpec,
+    genesis_storage: BTreeMap<Vec<u8>, Vec<u8>>,
```

Comment:
I don't like that we're not just borrowing the keys and values here, but I couldn't think of a better way of doing things that didn't make the struct self-referential.

Reply:
Can't you just modify `structs::ClientSpec`?

Reply:
Not easily, as it's a serde struct

Reply:
What's the problem? Are the serde traits not implemented on `BTreeMap`?

Reply:
Well the client spec is deserialized from the json right? So to create a BTreeMap inside that, you'd have to implement a custom deserialization or something.

Reply:
What I'm suggesting is to change [these two lines](https://github.com/paritytech/smoldot/blob/e980805d425f444e38972e6dd00d2028dd1b4750/src/chain_spec/structs.rs#L87-L88) to be a `BTreeMap` instead of `HashMap`